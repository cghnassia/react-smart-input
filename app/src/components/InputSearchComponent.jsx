import React from 'react'

class InputSearchComponent extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			searchString: ''
		};
	}

	onButtonClick() {
		//console.log(this.state)
		//this.setState({likesCount: this.state.likesCount + 1});
		this.props.onSearch(this.state.searchString);
	}

	onInputChange(event) {
		this.setState({
			searchString: event.target.value
		});
	}

	render() {
		return(
			<div className='input-search'>
				<input type="text" placeholder="search" onChange={this.onInputChange.bind(this)}/>
				<button onClick={this.onButtonClick.bind(this)}>Search!</button>
			</div>
		);
	}
}

export default InputSearchComponent;