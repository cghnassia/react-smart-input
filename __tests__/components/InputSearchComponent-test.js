import React from 'react';
import InputSearchComponent from '../../app/src/components/InputSearchComponent.jsx';
import renderer from 'react-test-renderer';
import TestUtils from 'react-addons-test-utils';

describe('Link', () => {
    it('renders correctly', () => {
        const tree = renderer.create(
            <InputSearchComponent />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

  /*it ('When clicking on search, binded function should be called', () => {
        let returnedValue = null;
        let onSearch = function(value) {
            returnedValue = value;
        }

        const inputSearchComponent = TestUtils.renderIntoDocument(
            <InputSearchComponent onSearch={onSearch} />
        );

        TestUtils.Simulate.change(
          TestUtils.findRenderedDOMComponentWithTag(inputSearchComponent, 'button')
        );

        expect(returnedValue).toEqual('');
    });*/
});