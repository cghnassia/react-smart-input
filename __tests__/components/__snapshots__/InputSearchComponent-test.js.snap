exports[`Link renders correctly 1`] = `
<div>
  <input
    onChange={[Function bound onInputChange]}
    placeholder="search"
    type="text" />
  <button
    onClick={[Function bound onButtonClick]}>
    Search!
  </button>
</div>
`;
